### DISCUSSION TITLE
arm_pack_manager

### DISCUSSION THREAD

**Discussion by JojoS62** [mbed-ce/mbed-os#88] [General]
I'm looking for the tools/arm_pack_manager, is this removed in mbed-ce? Where are the linker script symbols MBED_RAM_SIZE and MBED_RAM_START taken from? In mbed-os, these were created from the arm_pack_manager/index.json.

**Comment by multiplemonomials** [comment]
Hmm, I believe that the ram size and start come from a target's entry `target/targets.json`.

I'll be the first to admit I'm a bit fuzzy on exactly what these defines do and why only some targets have them, though.

**Comment by JojoS62** [comment]
I have digged into this a little further, I had overseen the first line in the linker scripts which includes the cmsis_nvic.h. With the cmake based cli, the linker .ld is first pre processed by the compiler and the RAM/ROM symbols are resolved. I guess the symbols in cmsis_nvic were created from the pack manager json file. In the past, the sizes were not always correct and not easy to maintain for the large number of targets. So the automatic process made sense.
